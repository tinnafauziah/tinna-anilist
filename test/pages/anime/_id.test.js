import flushPromises from 'flush-promises';
import detail from '~/pages/anime/_id';
import animeMock from '~/test/mocks/store-mocks/anime-mocks';

let store;
const mocks = {
  $router: {
    push: jest.fn(),
  },
  $route: {
    params: {
      id: 123
    }
  }
};
const modules = { ...animeMock };

describe('Anime Detail Page', () => {
  const mockPage = () => {
    store = global.createStore({ ...modules });
    return global.shallowMount(detail, { store, mocks });
  };
  it('should anime detail page match snapshot', async () => {
    const wrapper = mockPage();
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });
});
