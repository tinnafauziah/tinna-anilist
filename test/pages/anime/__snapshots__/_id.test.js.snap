// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Detail Page should anime detail page match snapshot 1`] = `
<div
  class="detail-anime"
>
  <v-layout-stub
    tag="div"
    wrap="wrap"
  >
    <v-flex-stub
      class="xs3 mr-3"
      tag="div"
    >
      <v-img-stub
        options="[object Object]"
        position="center center"
        src="https://s4.anilist.co/file/anilistcdn/media/anime/cover/large/bx127230-FlochcFsyoF4.png"
        transition="fade-transition"
      />
      <v-layout-stub
        align-center="align-center"
        tag="div"
        wrap="wrap"
      >
        <v-flex-stub
          class="xs1 mr-1"
          tag="div"
        >
          <v-icon-stub
            color="green"
          >
            mdi-emoticon-excited-outline
          </v-icon-stub>
        </v-flex-stub>
        <v-flex-stub
          class="xs10"
          tag="div"
        >
          <span
            class="subtitle-1"
          >
            86%
          </span>
        </v-flex-stub>
      </v-layout-stub>
    </v-flex-stub>
    <v-flex-stub
      class="xs6"
      tag="div"
    >
      <div
        class="title"
      >
        Chainsaw Man
      </div>
      <div
        class="subtitle-1"
      >
        Denji is a teenage boy living with a Chainsaw Devil named Pochita. Due to the debt his father left behind, he has been living a rock-bottom life while repaying his debt by harvesting devil corpses with Pochita.
        <br />
        <br />
        
One day, Denji is betrayed and killed. As his consciousness fades, he makes a contract with Pochita and gets revived as "Chainsaw Man" â€” a man with a devil's heart.
        <br />
        

        <br />
        
(Source: Crunchyroll)
      </div>
    </v-flex-stub>
  </v-layout-stub>
</div>
`;
